server/src/main/java/org/opensearch/index/translog/RemoteFsTranslog.java

```diff
@@ -374,9 +375,9 @@ private boolean prepareAndUpload(Long primaryTerm, Long generation) throws IOExc
         // ReplicationTracker.primaryMode() as true. However, before we perform the `internal:index/shard/replication/segments_sync`
         // action which re-downloads the segments and translog on the new primary. We are ensuring 2 things here -
         // 1. Using startedPrimarySupplier, we prevent the new primary to do pre-emptive syncs
-        // 2. Using syncPermits, we prevent syncs at the desired time during primary relocation.
-        if (startedPrimarySupplier.getAsBoolean() == false || syncPermit.tryAcquire(SYNC_PERMIT) == false) {
-            logger.debug("skipped uploading translog for {} {} syncPermits={}", primaryTerm, generation, syncPermit.availablePermits());
+        // 2. Using syncPermit, we prevent syncs at the desired time during primary relocation.
+        if (startedPrimarySupplier.getAsBoolean() == false || syncPermit.isHeldByCurrentThread() || syncPermit.tryAcquire() == null) {
+            logger.debug("skipped uploading translog for {} {} syncPermit={}", primaryTerm, generation, availablePermits());
```

Comment:
I realize this preserves existing behavior, but I'm wondering if it wouldn't be more clear with just a true/false value, as `availablePermits()` doesn't have much meaning any more and just adds an extra step of converting to an integer.